hollow/src/main/java/com/netflix/hollow/api/producer/ListenerSupport.java

```diff
@@ -70,128 +74,260 @@ void remove(IncrementalCycleListener listener) {
     }
 
     void fireProducerInit(long elapsedMillis) {
-        for(final HollowProducerListener l : listeners) l.onProducerInit(elapsedMillis, MILLISECONDS);
+        for(final HollowProducerListener l : listeners) {
+            try {
+                l.onProducerInit(elapsedMillis, MILLISECONDS);
+            } catch(RuntimeException e) {
+                log.warning("Execution of " + l.getClass().getName() + "  during fireProducerRestoreStart failed | " + e);
```

Comment:
Let's use the [version](https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html#log(java.util.logging.Level,%20java.lang.String,%20java.lang.Throwable)) that takes a Throwable instead, so we can get stack traces.
At that point we could even simplify the message to something like "Error executing listener", since we'll have enough details in the stack trace.